db/system_keyspace.cc

```diff
@@ -82,6 +82,7 @@ namespace {
             system_keyspace::GROUP0_HISTORY,
```

Comment:
Schema commitlog is more durable then regular one because it is always in `db::commitlog::sync_mode::BATCH` mode. In this mode writes wait for data to actually be on disk at least in commitlog before returning.

Reply:
Ok, but given that the tables we're dealing with were using wait_for_sync_to_commitlog -- we didn't gain any durability, right?

Reply:
If this is the case then right as far as I see.

Reply:
Yes, in fact that's what I do on my branch which sparked this pull request. Please feel free to add to this series or do a follow up.

Reply:
> I don't see why it should be different for schema commit log.

`table::flush` does `fsync` on the just written sstables (`scylladb/sstables/mx/writer.cc/writer::close_data_writer`), writing to schema commitlog without 
`use_o_dsync` won't `fsync`, the data can be lost in case of power outage, so this commit worsens durability.

Reply:
But does this schenario (write to schema commitlog without use_o_dsync) ever happen? apart from developer/test settings.

Reply:
even without o_dsync we do sync, batch_cycle() will call sync(), which calls _file.flush(), which calls fdatasync().  

```

future<>
posix_file_impl::flush() noexcept {
    if ((_open_flags & open_flags::dsync) != open_flags{}) {
        return make_ready_future<>();
    }
    return engine().fdatasync(_fd);
}

```

Reply:
> even without o_dsync we do sync, batch_cycle() will call sync(), which calls _file.flush(), which calls fdatasync().

I didn't notice that, thank you. There is no problem then, I suppose.

Reply:
> But does this schenario (write to schema commitlog without use_o_dsync) ever happen? apart from developer/test settings.

`use_o_dsync` is accessible through `config.cc`, so the use could set it to `false`.